front/pages/api/v1/w/[wId]/spaces/[spaceId]/data_sources/[dsId]/folders/index.ts

```diff
@@ -39,6 +35,18 @@ async function handler(
 
   const dataSource = await DataSourceResource.fetchById(auth, dsId);
 
+  let { spaceId } = req.query;
```

Comment:
could that part be done via the withResource wrapper? (auto fetch space with right checks)